src/main/java/com/jagrosh/jmusicbot/commands/dj/PlaynowCmd.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2018 John Grosh <john.a.grosh@gmail.com>.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jagrosh.jmusicbot.commands.dj;
+
+import com.jagrosh.jdautilities.command.CommandEvent;
+import com.jagrosh.jmusicbot.Bot;
+import com.jagrosh.jmusicbot.audio.AudioHandler;
+import com.jagrosh.jmusicbot.audio.QueuedTrack;
+import com.jagrosh.jmusicbot.commands.DJCommand;
+import com.jagrosh.jmusicbot.utils.FormatUtil;
+import com.sedmelluq.discord.lavaplayer.player.AudioLoadResultHandler;
+import com.sedmelluq.discord.lavaplayer.tools.FriendlyException;
+import com.sedmelluq.discord.lavaplayer.track.AudioPlaylist;
+import com.sedmelluq.discord.lavaplayer.track.AudioTrack;
+import net.dv8tion.jda.core.entities.Message;
+
+/**
+ *
+ * @author Daniel Liljeberg (liljebergxyz@gmail.com)
+ */
+public class PlaynowCmd extends DJCommand
+{
+    private final String loadingEmoji;
+    
+    public PlaynowCmd(Bot bot, String loadingEmoji)
+    {
+        super(bot);
+        this.loadingEmoji = loadingEmoji;
+        this.name = "playnow";
+        this.arguments = "<title|URL>";
+        this.help = "plays a single song now";
+        this.beListening = true;
+        this.bePlaying = false;
+    }
+    
+    @Override
+    public void doCommand(CommandEvent event)
+    {
+        if(event.getArgs().isEmpty() && event.getMessage().getAttachments().isEmpty())
+        {
+            event.replyWarning("Please include a song title or URL!");
+            return;
+        }
+        AudioHandler handler = (AudioHandler)event.getGuild().getAudioManager().getSendingHandler();
+        event.reply(event.getClient().getSuccess()+" Skipped **"+handler.getPlayer().getPlayingTrack().getInfo().title+"**");
+        String args = event.getArgs().startsWith("<") && event.getArgs().endsWith(">") 
+                ? event.getArgs().substring(1,event.getArgs().length()-1) 
+                : event.getArgs().isEmpty() ? event.getMessage().getAttachments().get(0).getUrl() : event.getArgs();
```

Comment:
This is quite compact, but i feel like its too dense. I'm able to understand it, but it can be hard to read.

Reply:
This is actually straight up copy-pasted from a different command. But I could clean it up while I'm at it if you'd like